Cargo.toml

```diff
@@ -45,8 +45,11 @@ tokio = { version = "1.38.0", features = [
 toml_edit = { version = "0.22.15", features = ["serde"] }
 uuid = { version = "1.10.0", features = ["serde", "v4"] }
 yansi = "1.0.1"
-zip = { version = "2.1.3", default-features = false, features = ["deflate"] }
-zip-extract = "0.1.3"
+zip = { version = "2.1.3", default-features = false, features = [
+    "deflate",
+    "bzip2",
```

Comment:
Do we need bzip2, knowing we only use this lib to create zip archives?

Reply:
But we use `zip-extract` for that, and it doesn't have a feature flag for bzip2.

Reply:
Also, at that point why not add the other compression techniques like flate2, zlib, lzma etc?

Reply:
zip-extract is a thin wrapper around zip

Reply:
I understand, I find it just a bit weird to enable this feature via the `zip` crate directly and not `zip-extract`. If one day zip-extract uses a different major version of `zip` compared to the one we use, then that feature would not be enabled for extraction anymore.
Regarding my other questions, why add bzip2 and not flake2, zlib, lzma etc?

Reply:
I've added support upstream and added in https://github.com/mario-eth/soldeer/pull/161/commits/5ee2b3e592153c8ba662fe936b15d2e8835112ae, in any case if the versions diverge it duplicates zip again which is not good

There's no need if the only one used is bzip2, we only need to extract what we compressed ourselves

Reply:
> There's no need if the only one used is bzip2, we only need to extract what we compressed ourselves

That's not strictly true, users can pass a custom URL for their dependencies, and wee need to decompress the result archive. This could potentially be any format supported by `zip` if we add the relevant flags. So it's a decision we can make: do we want to support URLs with files only containing zip/deflate, or do we want to support more archive formats. Still don't know where bzip2 comes into play, sorry for my confusion :confused:

Reply:
Thanks for fixing the upstream flags though! That's nice to have the same options as `zip`

Reply:
Oh okay my bad, I didn't know one test relied on bzip2 and that it was mentioned in the manpage. the `flate` feature flag covers `tar.gz` right? If so then I think I'm okay with the current choice!